### PR TITLE
Require sudo mode on the provisioning API

### PR DESCRIPTION
**Pull request by nickvergessen**
Ref #2487 

@LukasReschke @rullzer

### CODE REVIEW

File: apps/provisioning_api/lib/Controller/GroupsController.php [hunk 1]
```diff
@@ -128,7 +128,7 @@ public function getGroup($groupId) {
 	/**
 	 * creates a new group
 	 *
-	 * @NoAdminRequired
```
Comment (nickvergessen) [inline apps/provisioning_api/lib/Controller/GroupsController.php]: @rullzer you added that by accident in https://github.com/nextcloud/server/commit/432e7c93c6dad564abbaec1e3d374f73653d7ba6

This allows all users to create groups Oo

Comment (rullzer) [inline apps/provisioning_api/lib/Controller/GroupsController.php]: Ah mmm my bad

Comment (LukasReschke) [inline apps/provisioning_api/lib/Controller/GroupsController.php]: @nickvergessen ❤️

**Review by LukasReschke — APPROVED** [review]

**Comment by LukasReschke** [comment]
>error_outlineERROR: 500 Internal Server Error: write /docker/aufs/layers/8d99f89faa9f651a88d3d69ce07cb8917cba2652143e34ce37d0bf961746fe66-init: no space left on device

Let's retrigger the tests.

cc @MorrisJobke

**Comment by MorrisJobke** [comment]
> Let's retrigger the tests.

:(

**Comment by MorrisJobke** [comment]
Provisioning API integration tests say 💥

**Comment by MorrisJobke** [comment]
> Provisioning API integration tests say 💥

And I also think that this will cause a lot of broken scripts - we should add this to the release notes to avoid trouble here and guiding people towards the app tokens ;)

**Comment by rullzer** [comment]
Mmmm why do they 💥 ? Because we use basic auth on OCS... or does do the intergration tests use dark cookie magic?

**Comment by LukasReschke** [comment]
> And I also think that this will cause a lot of broken scripts - we should add this to the release notes to avoid trouble here and guiding people towards the app tokens ;)

Scripts should use Basic Auth as specified in the OCS specs. :)

**Comment by MorrisJobke** [comment]
@LukasReschke Ah that means HTTP Auth is fine for this. I thought it needs to be an app specific password to not catch the password confirmation ... got it :)

**Comment by nickvergessen** [comment]
Will have a look at the tests

**Comment by nickvergessen** [comment]
Seems like we dont use basic auth, but changing it doesn't work for me either. Also my xdebug fails to debug those tests, so maybe someone else can have a look

**Comment by MorrisJobke** [comment]
> Seems like we dont use basic auth, but changing it doesn't work for me either. Also my xdebug fails to debug those tests, so maybe someone else can have a look

cc @rullzer ?

**Comment by LukasReschke** [comment]
Will take a look.

**Comment by rullzer** [comment]
Fixed. @LukasReschke please verify my change

**Comment by LukasReschke** [comment]
Nice, even with tests 🚀

**Comment by LukasReschke** [comment]
LGTM

**Comment by MorrisJobke** [comment]
:+1:

**Pull request merged**